rxandroidble/src/main/java/com/polidea/rxandroidble2/scan/ScanFilter.java

```diff
@@ -278,7 +275,7 @@ public boolean matches(RxBleInternalScanResult scanResult) {
 
         // Local name match.
         if (mDeviceName != null) {
-            if (!(mDeviceName.equals(scanRecord.getDeviceName()) || mDeviceName.equals(device.getName()))) {
+            if (!mDeviceName.equals(scanRecord.getDeviceName())) {
```

Comment:
This change drops comparison with the device name. However, I believe this is the correct behaviour. The name in the advertisement could be different to the name read from the device and cached (I guess that's where this device name comes from, if it isn't just the name from the scan record)

Reply:
Is this what Android BLE ScanFilter does?
I suppose this would be a breaking change for many implementations. I'll change it back

Reply:
I have reverted the change, if you like the idea of `setDeviceName(String deviceName, boolean onlyScanRecord)` perhaps I should do it in another PR.